### PR TITLE
Onboard to scheduled LGTM run

### PR DESCRIPTION
**Pull request by TianqiZhang**
###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/docfx/pull/7129)

### CODE REVIEW

File: azure-pipelines-LGTM.yml
Comment (yufeih) [inline azure-pipelines-LGTM.yml:43]: Is there a way to get rid of this token?

Comment (yufeih) [reply azure-pipelines-LGTM.yml:67]: How about `$(System.AccessToken)`

File: azure-pipelines-LGTM.yml
Comment (yufeih) [inline azure-pipelines-LGTM.yml:29]: nit:
```suggestion
- job: Security Checks
```

File: azure-pipelines-LGTM.yml
Comment (TianqiZhang) [reply azure-pipelines-LGTM.yml:73]: In communication with security team, Denis Levin wrote:
> LGTM Proxy does support GitHub, but since the source code repository is at GitHub and not in ADO, and ADO Pipeline’s SYSTEM_ACCESSTOKEN we receive in LGTM proxy as a parameter is created at ADO, we can’t use it to authenticate against GitHub. Instead (currently) we use this token to check if it has read access to https://dev.azure.com/microsoft account. 
> At the moment, the only fix we can offer is to set up your pipeline with another token (PAT token) scoped to https://dev.azure.com/microsoft that have read access (code_read) to dev.azure.com/Microsoft and pass this PAT token to pipeline:
SYSTEM_ACCESSTOKEN: $(MICROSOFT_PAT_TOKEN)

> We have an alternative authentication method in test environment that will take care of this issue, but it’s a few days away from deployment to production.

So we should be able to get rid of this token in a few days.

**Review by yufeih — APPROVED** [review]

**Comment by herohua** [comment]
Is there a particular reason why we'd like to have this as a separate pipeline (yml)?

**Comment by yufeih** [comment]
> Is there a particular reason why we'd like to have this as a separate pipeline (yml)?

It took very long to run LGTM (20+ minutes), a dedicated pipeline allows us to run these as an optional check that does not block normal workflow. (It is configured as per commit, but I see no problem running LGTM once every day).

**Pull request merged**
